lib/fill/index.js

```diff
@@ -0,0 +1,13 @@
+import curryN from '../curryN'
+import toArray from '../toArray'
+
+export default curryN(2, function fill(val, start, end, arr) {
+  var last = arguments[arguments.length - 1]
+  return Array.isArray(last)
+    ? val
+      ? toArray(last).fill(val, Array.isArray(start) ? undefined : start, Array.isArray(end) ? undefined : end)
```

Comment:
repepetive code, extract it to a function

Reply:
agree

Reply:
Validate it as a number only, why do you need to check is it's an array?